x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -494,4 +494,47 @@ export class ManifestManager {
   public getArtifactsClient(): EndpointArtifactClientInterface {
     return this.artifactClient;
   }
+
+  /**
+   * Cleanup .fleet-agents indice if there are some orphan artifacts
+   */
+  public async cleanup(manifest: Manifest) {
+    try {
+      const fleetArtifacts = await this.artifactClient.listArtifacts();
+
+      if (isEmpty(fleetArtifacts)) {
+        return;
+      }
+
+      const artifactsToDelete = [];
+      const manifestArtifactsIds = manifest
+        .getAllArtifacts()
+        .map((artifact) => getArtifactId(artifact));
+
+      for (const fleetArtifact of fleetArtifacts) {
+        const artifactId = getArtifactId(fleetArtifact);
+        const isArtifactInManifest = manifestArtifactsIds.includes(artifactId);
+
+        if (!isArtifactInManifest) {
+          this.logger.info(`Artifact ${artifactId} needs to be removed because it's orphan`);
```

Comment:
Suggestion: remove this logger call and log the data below only (on line 528) so that we are not that chatty